DEV_GUIDE.md

```diff
@@ -0,0 +1,152 @@
+# PlayerUI Typescript Development Guide
+
+### Sections:
+1. [Setup](#Setup)
+2. [Adding a new plugin](#adding-a-new-plugin)
+
+## Setup
+### Demo Application
+The demo app can be built and launched from the command line with bazel:
+```bash
+bazel build //docs/site:site
+bazel run //docs/site:site
+```
+
+## Adding a new plugin
+Adding a new plugin is simple using predefined macros for `BUILD.bazel` files.
+
+### Core vs React Plugins
+Core plugins provide shared functionality, usable on all platforms.
+
+React plugins add react specific functionality.
+
+### Scaffold
+All plugins follow a common structure:
+
+The basic structure is as follows:
```

Comment:
I would expand a bit on what that common structure is and why its important to make sure since it might not be immediately understandable by looking at the diagram